File: skills/ai/optverse/huawei-cloud-optv-evolve-management/templates/python/evaluator.py
"""
Evaluator function template - Python (minimal workable version)

Constraints:
- Must be a parameterless `def evaluate()`; the filename must exactly match
  the value passed to `CreateEvolveTask --evaluator_file`
- `evaluate()` returns a single float; larger / smaller is determined by
  `--evaluator_parameter.smaller_better`

The platform imports the candidate code and the candidate function, so this
function only needs to:
1. Prepare the test data
2. Call the function under evaluation directly
3. Return a float score
"""


def evaluate():
    import time
    import random

    # 1. Prepare the test data
    test_data = [random.randint(1, 10_000_000) for _ in range(100_000)]

    # 2. Call the function under evaluation (the platform auto-injects
    #    the function being evolved, `sort_algorithm`)
    start = time.time()
    sort_algorithm(test_data)
    elapsed = time.time() - start

    # 3. Return the score (e.g. elapsed time; interpreted with
    #    `smaller_better=true`)
    return elapsed

File: skills/ai/optverse/huawei-cloud-optv-evolve-management/templates/cpp/evaluator.cpp
// evaluator.cpp - C++ evaluator command source
//
// How the evolution platform calls this:
//   1. `CreateAlgorithm --build_command="bash ./build.sh"` compiles first
//   2. In `CreateEvolveTask`, `evaluator_func_name` points directly at the
//      compiled artefact (e.g. "build/evaluate <args>"); `evaluator_file` is
//      left blank
//   3. Once the candidate `sort_algorithm.cpp` is ready, the platform launches
//      this command
//
// Constraints:
//   - The command MUST print the final score (a single float) on its own line
//     to stdout
//   - Do NOT print anything else to stdout (route logs to stderr)
//   - The algorithm source file MUST wrap the function under evolution with
//     `// EVOLVE_START` / `// EVOLVE_END` markers
//
// This file is a minimal template; replace the `value` calculation with your
// own business logic.

#include <cstdio>
#include <cstdlib>
#include <string>

int main() {
    // TODO: invoke the candidate `sort_algorithm` and compute the evaluation
    // score per the protocol
    // Typical shape: run the candidate binary / read stdin protocol / measure
    // time / print the score
    double value = 0.0;

    std::printf("%.6f\n", value);
    return 0;
}
